### PR TITLE
gh-1360 REST: Ensure actuator endpoints are secured with OAuth

### PR DESCRIPTION
**Pull request by ghillert**
* Add `@EnableWebSecurity` to `OAuthSecurityConfiguration`
* Add integration tests

resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/1360

### CODE REVIEW

File: spring-cloud-starter-dataflow-server-local/src/test/resources/org/springframework/cloud/dataflow/server/local/security/oauthConfig.yml [hunk 1]
```diff
@@ -1,6 +1,3 @@
-management:
-  security:
-    enabled: true
```
Comment (ilayaperumalg) [inline spring-cloud-starter-dataflow-server-local/src/test/resources/org/springframework/cloud/dataflow/server/local/security/oauthConfig.yml]: I see that this is removed as it requires `ADMIN` role (from spring-boot) to access the management actuator endpoints. I understand this is only in tests, but, what happens when `management.security.enabled=true` at the server level?

**Comment by trisberg** [comment]
Not sure I see where the original issue was, could you explain what you are trying to fix?

**Comment by ghillert** [comment]
@trisberg One requirement ( @sabbyanandan ) by an earlier issue was that OAuth2 shall not require the presence of `security.basic.enabled`. It turns out that we still had some Boot magic going on that was affected by this and it turned out that all of a sudden the management endpoints were unprotected. That's why I added the `@EnableWebSecurity` to have all security being driven by `OAuthSecurityConfiguration`.

File: spring-cloud-starter-dataflow-server-local/src/test/resources/org/springframework/cloud/dataflow/server/local/security/oauthConfig.yml (hunk 1 above)
Comment (ghillert) [inline spring-cloud-starter-dataflow-server-local/src/test/resources/org/springframework/cloud/dataflow/server/local/security/oauthConfig.yml]: Interesting point I will add a test suite with that enabled to have that test-covered. I will add that once I brought the kids to school.

Comment (trisberg) [inline spring-cloud-starter-dataflow-server-local/src/test/resources/org/springframework/cloud/dataflow/server/local/security/oauthConfig.yml]: If `management.security.enabled=true` you'll get an error from Boot saying:
```
"Access is denied. User must have one of the these roles: ADMIN"
```
when accessing `/management` endpoints

**Comment by trisberg** [comment]
Rebased and merged as commit 092a00240dfe27737177ef4f5024491eb2b11877

**Pull request closed without merging**